doc/introduction/measurements.rst

```diff
@@ -354,6 +354,53 @@ The transform can be applied to a QNode as follows:
     If the ``defer_measurements`` transform is used in analytic mode, ``backprop`` is also a viable
     option.
 
+.. _tree_traversal:
+
+The tree-traversal algorithm
+****************************
+
+Dynamic circuit execution is akin to traversing a binary tree where each MCM
+corresponds to a node and groups of gates between the MCMs correspond to edges.
+The :func:`~.pennylane.dynamic_one_shot` approach picks a branch of the tree randomly
+and simulates it from beginning to end.
+This is wasteful in many cases; the same branch is simulated many times
+when there are more shots than branches for example.
+The tree-traversal algorithm does away with such redundancy while retaining the
+exponential gains in memory of the one-shot approach compared with the deferred
+measurement principle, among other advantages.
+
+Briefly, it proceeds from the top of the tree and begins splitting the circuit
+into two: a circuit running to the next MCM replacing the MCM by a sample
+measurement, and a circuit consisting of the rest of the circuit. A key point is that
```

Comment:
I'm not sure I followed the first sentence. Though admittedly, this algorithm is always going to be hard to describe in just one paragraph!

Reply:
How about 
```
Briefly, it proceeds cutting an :math:`n_{MCM}` circuit into :math:`n_{MCM}+1`
circuit segments. Each segment can be executed on either the `0`- or `1`-branch,
which gives rise to a binary tree with :math:`2^{n_{MCM}}` leaves.  Terminal
measurements are obtained at the leaves, and propagated and combined back up at each
node up the tree. The tree is visited using a depth-first pattern. The tree-traversal
method improves on :func:`~.pennylane.dynamic_one_shot` by taking all samples at a
node or leaf at once. Neglecting overheads, simulating all branches requires the same
amount of computations as :func:`~. pennylane.defer_measurements`, but without the
:math:`O(2^{n_{MCM}})` memory requirement.  To save time, a copy of the state vector
is made at every branching point, or MCM, requiring at most :math:`n_{MCM}+1` state
vectors at any instant, an exponential improvement compared with :func:`~. pennylane.
defer_measurements`.  Since the counts of many nodes come out to be zero in practice,
it is often possible to ignore entire sub-trees, thereby reducing the computational
burden.
```
?

Reply:
@mudit2812 Do you prefer this one too?

Reply:
Yeah the suggested explanation is very nice!